pkg/syncer/upsync/upsync_reconcile.go

```diff
@@ -45,78 +42,34 @@ const (
 	ResourceVersionAnnotation = "workload.kcp.io/rv"
 )
 
-func (c *controller) reconcile(ctx context.Context, upstreamObject *unstructured.Unstructured, gvr schema.GroupVersionResource, upstreamClusterName logicalcluster.Name, upstreamNamespace, upstreamName string, dirtyStatus bool) (bool, error) {
-	logger := klog.FromContext(ctx)
+type reconciler struct {
+	cleanupReconciler
```

Comment:
compare other controller with multiple reconciler. We have an established pattern for that.

Reply:
I know but I did so since here there is common code between the cleanup reconciler and the main one.